2008-06-08  Kristian Rietveld  <kris@imendio.com>

	Merge from trunk:

	* gtk/gtktreeviewcolumn.c (gtk_tree_view_column_update_button):
	always determine what arrow to show if show_sort_indicator is true.
	(#352738, Chris Vine).

diff --git a/gtk/gtktreeviewcolumn.c b/gtk/gtktreeviewcolumn.c
index 9331b8f..52994cf 100644
--- a/gtk/gtktreeviewcolumn.c
+++ b/gtk/gtktreeviewcolumn.c
@@ -925,8 +925,7 @@ gtk_tree_view_column_update_button (GtkTreeViewColumn *tree_column)
 					  &sort_column_id,
 					  NULL);
 
-  if (tree_column->show_sort_indicator
-      && (sort_column_id >= 0 && sort_column_id == tree_column->sort_column_id))
+  if (tree_column->show_sort_indicator)
     {
       gboolean alternative;
 
-- 
1.5.4.5

